packages/widget/src/components/transfer/fungible/transfer-detail/transfer-detail.ts

```diff
@@ -0,0 +1,85 @@
+import { FeeHandlerType } from '@buildwithsygma/sygma-sdk-core';
+import type {
+  BaseConfig,
+  EvmFee,
+  Network,
+  Resource
+} from '@buildwithsygma/sygma-sdk-core';
+import '../../../common/buttons/button';
+import type { HTMLTemplateResult } from 'lit';
+import { html } from 'lit';
+import { when } from 'lit/directives/when.js';
+import { customElement, property } from 'lit/decorators.js';
+import { tokenBalanceToNumber } from '../../../../utils/token';
+import { BaseComponent } from '../../../common/base-component';
+import { styles } from './styles';
+
+@customElement('sygma-fungible-transfer-detail')
+export class FungibleTransferDetail extends BaseComponent {
+  static styles = styles;
+
+  @property({ type: Object })
+  fee?: EvmFee;
+
+  @property({ type: Object })
+  selectedResource?: Resource;
+
+  @property({ type: Object })
+  sourceDomainConfig?: BaseConfig<Network>;
+
+  displayFee(): string {
+    if (!this.fee) return '';
+    let balance = '';
+    let symbol = '';
+    let decimals: number | undefined = undefined;
+    const { type, fee } = this.fee;
+
+    if (type === FeeHandlerType.PERCENTAGE) {
+      if (this.selectedResource) {
+        if (this.selectedResource.symbol) {
+          symbol = this.selectedResource.symbol;
+        }
+
+        if (this.selectedResource.decimals) {
+          decimals = this.selectedResource.decimals;
```

Comment:
It's probably better to separate the getter of fee details and the display, f.e:
```
 getFeeDetails(): { symbol: string; decimals?: number } {
    switch (this.fee?.type) {
      case FeeHandlerType.PERCENTAGE:
        return {symbol: '', decimal: ''}; // logic is missing
      case FeeHandlerType.BASIC:
        return {symbol: '', decimal: ''}; // logic is missing
      default:
        return { symbol: '', decimals: undefined };
    }
  }
```